### ISSUE TITLE
[Question / Feature Request] Are there plans of making timeboxExpression a public feature?

### ISSUE THREAD

**Issue by slisaasquatch** [jsonata-js/jsonata#521]
What I'm talking about is this method that's currently only used in internal tests
https://github.com/jsonata-js/jsonata/blob/97295a6fdf0ed0df7677e5bf36a50bb633eb53a2/test/run-test-suite.js#L158

I can see this being very useful in a lot of use cases, and it's certainly useful for me. Right now I'm using the same source code to set a maximum timeout and a max depth, and I'm wondering whether this will ever become a public feature and whether I can rely on things like `__evalute_entry` and `__evaluate_exit` going forward since those are not in the public docs.

Thanks!

**Comment by andrew-coleman** [comment]
I put this in to prevent infinite loops in the exerciser at the start of the project when there was no other way of interrupting an expression.  But since then, the evaluator was rewritten to use generators so it's unlikely that this earlier hack will become a public feature.  Having said that, I have no plans to take it away anytime soon.  Not sure if that helps.

**Comment by slisaasquatch** [comment]
Yes that helps. So are there plans of implementing a better alternative of this as a public feature?

**Comment by tbinna** [comment]
@andrew-coleman would you mind explaining a bit more (or point me to a resource) about how timeboxing and call stack limitation can be achieved with the generator-based implementation?

The way I read your comment is that you are saying using `__evalute_entry` and `__evalute_exit` callbacks is not necessary anymore and the same can be achieved in a different/better way, or did I understand that wrongly?

Thanks!
